### PR TITLE
Layermetadata

### PR DESCRIPTION
**Pull request by KaiVolland**

### CODE REVIEW

File: src/main/java/de/terrestris/momo/web/MetadataController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	private MetadataService metadataService;
+
+	@RequestMapping(value = "/csw.action", method = {RequestMethod.POST})
+	public @ResponseBody Map<String, Object> cswRequest(@RequestParam String xml){
+		try {
+			String decodedXML = java.net.URLDecoder.decode(xml, "UTF-8");
+			String response = this.metadataService.cswRequest(decodedXML);
+			return ResultSet.success(response);
+		} catch (URISyntaxException | HttpException | IOException e) {
+			return ResultSet.error("Error while updating Sld: " + e.getMessage());
```
Comment (weskamm) [inline src/main/java/de/terrestris/momo/web/MetadataController.java]: wrong msg

File: src/main/java/de/terrestris/momo/web/MetadataController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class MetadataController {
+
+	@Autowired
+	@Qualifier("metadataService")
+	private MetadataService metadataService;
+
+	@RequestMapping(value = "/csw.action", method = {RequestMethod.POST})
+	public @ResponseBody Map<String, Object> cswRequest(@RequestParam String xml){
+		try {
+			String decodedXML = java.net.URLDecoder.decode(xml, "UTF-8");
```
Comment (weskamm) [inline src/main/java/de/terrestris/momo/web/MetadataController.java:36]: could be done in service method

**Comment by weskamm** [comment]
just added 2 minors, else good 2 go

File: src/main/java/de/terrestris/momo/web/MetadataController.java (hunk 2 above)
Comment (KaiVolland) [reply src/main/java/de/terrestris/momo/web/MetadataController.java:36]: I'd like to keep this as it is, as the service method expects an decoded and valid xmlString.

Comment (marcjansen) [reply src/main/java/de/terrestris/momo/web/MetadataController.java:36]: I am a bit suprised you need to decode at all. This does no harm,  but nonetheless I am surprised.

**Pull request merged**